### ISSUE TITLE
Wrong data structure

### ISSUE THREAD

**Issue by Sresht** [APeters97/CS-435-Project-2#11]
ArrayLists are used to indicate a list of things where the order is important. list.contains(item) takes O(n) time because the order matters. 

Sets are used to indicate a list of things where the order doesn't matter. set.contains(item) takes O(1) time as a result.

Using an ArrayList where you should be using a HashSet is not only inefficient, but it's also incredibly confusing to read because it makes other developers wonder what the order of your nodes is.

https://github.com/APeters97/CS-435-Project-2/blob/daa2caff61a24b019356b191a49b534139afc7e9/Graph.java#L9

**Comment by APeters97** [comment]
This is true that the ordering of the vertices and neighbor lists don't matter, but I am using an ArrayList functionality. The way I wrote my code, for the functions that require us to create a random graph with randomly assigned edges, when I am on a current node I generate a random number from 1-n (the number of nodes) and that represents the node I am going to add an edge to. So I have to use the ArrayList get() function to grab that node from the vertices list.
